src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -181,9 +181,7 @@ CreateInsertSelectPlan(Query *originalQuery,
 	deferredError = ErrorIfOnConflictNotSupported(originalQuery);
 	if (deferredError != NULL)
 	{
-		distributedPlan = CitusMakeNode(DistributedPlan);
-		distributedPlan->planningError = deferredError;
-		return distributedPlan;
+		RaiseDeferredError(deferredError, ERROR);
```

Comment:
maybe add a comment on why we're raising here, since it's not the common thing to do